### DISCUSSION TITLE
different si-snr of valid set in training and scoring phase

### DISCUSSION THREAD

**Discussion by 1215thebqtic** [espnet/espnet#3816] [Q&A]
Hi,

I trained a denoising model (1 speaker) using conv-tasnet with si-snr loss. In training phase, a valid si-snr value is calculated when one epoch is finished, and the number will be recorded in train.log. After the training phase, a scoring phase will calculate si-snr on the enhanced dev set. I found that the si-snr values are different in the two phases. In training phase, the si-snr of the train set is positive(~10), but the valid si-snr is negative(~-2). In the final scoring stage, si-snr is positive (~ 10). 

In addition, when I use mask mse loss (stft-rnn-istft) to train the model, the two si-snr values in the training and scoring phase are the same.

I'm wondering anyone has the same problem? 

Thanks in advance.

**Comment by 1215thebqtic** [comment]
After checking the code, I think it's because of the different normalization methods of the output wav in training and inferencing. In training, the normalization is zero-mean(espnet2/enh/espnet_model.py, si_snr_loss_zeromean()), and in inferencing, wave samples will be normalized to [-1,1](espnet2/bin/enh_inference.py, normalize_output_wav).

**Reply by Emrys365** [reply]
Another possibility is that the chunk data loader is used during training, which may cause some mismatch between training and inference stages.
